netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -814,9 +826,16 @@ net_ebpf_ext_sock_addr_register_providers()
 void
 net_ebpf_ext_sock_addr_unregister_providers()
 {
-    for (int i = 0; i < NET_EBPF_SOCK_ADDR_HOOK_PROVIDER_COUNT; i++)
-        net_ebpf_extension_hook_provider_unregister(_ebpf_sock_addr_hook_provider_context[i]);
-    net_ebpf_extension_program_info_provider_unregister(_ebpf_sock_addr_program_info_provider_context);
+    for (int i = 0; i < NET_EBPF_SOCK_ADDR_HOOK_PROVIDER_COUNT; i++) {
+        if (_ebpf_sock_addr_hook_provider_registered[i]) {
+            net_ebpf_extension_hook_provider_unregister(_ebpf_sock_addr_hook_provider_context[i]);
+            _ebpf_sock_addr_hook_provider_registered[i] = false;
```

Comment:
should this set _ebpf_sock_addr_hook_provider_context[i] back to NULL?

Reply:
Fixed.